lib/CL/pocl_ipc_mutex_win.cc

```diff
@@ -0,0 +1,111 @@
+/* pocl_ipc_mutex.h: Named interprocess mutex that does not cause deadlock if a
+   process dies or exits without unlocking.
+
+   Copyright (c) 2024 Henry Linjamäki / Intel Finland Oy
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+*/
+
+#include "pocl_debug.h"
+#include "pocl_ipc_mutex.h"
+
+#include <windows.h>
+
+#include <cassert>
+#include <string>
+
+extern "C" int pocl_ipc_mutex_create(const char *Name,
```

Comment:
extern "C" is enough to add to the declaration. No need to add them for the definitions too.

https://stackoverflow.com/questions/1380829/is-extern-c-only-required-on-the-function-declaration

Reply:
Addressed.